ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala

```diff
@@ -197,16 +200,19 @@ object Update {
 
   /** Signal that a command submitted via [[WriteService]] was rejected.
     *
-    * See the different [[RejectionReason]] for why a command can be
-    * rejected.
+    * @param cancelled If false, the [[ReadService]]'s deduplication and submission rank guarantees
+    *   apply to this rejection. The participant state implementations should
+    *   strive to set this flag to false as often as possible so that applications
+    *   get better guarantees.
     */
   final case class CommandRejected(
       recordTime: Timestamp,
       submitterInfo: SubmitterInfo,
-      reason: RejectionReason,
+      reason: com.google.rpc.status.Status,
+      cancelled: Boolean,
```

Comment:
We probably don't want `cancelled` be a field of the class here, given that the ledger API now models this as a `definite_answer` key in the error details of `reason`. I'll make it derived.